### DISCUSSION TITLE
Query Editor and Ordering results

### DISCUSSION THREAD

**Discussion by joomlabeat** [Sequel-Ace/Sequel-Ace#1008] [Ideas]
Hi. When writing a query in the editor and got our results, there are so many times that I want to order the returned found-set of the query that I just run. 
In Sequel pro, it was possible to just alter the order of the returned and visible records by clicking on the column label and it would re-order the existing found-set, without passing the new order and perform a complete new query with that order (resulting in new results). 
That was a handy feature in many ways - when dealing with data and need various ways to visualise what is returned with each query.
Is it possible to do the same with Sequel Ace? It seems that every time I click a column label, it performs a whole new query (changing the ORDER of my main written query in the editor) and returns new results. I guess this is somewhat useful in its way, but in any case if I needed a complete different order when writing queries in the editor, I would have written that order in the first place. 
Any chance that this is adjustable somewhere in the preferences?

**Comment by Jason-Morcos** [comment]
Really depends on whether or not there's a limit on the resulting query (either in the query itself or from Sequel Ace). You can't change the order column on a limited query without making a second query because we can't tell if some records that should be sorted early are not in the initial result set. That's a little confusing, but does it make sense?
My hunch is Ace might be smart and not make a second query for result sets < the limit of returned results, but if not I think that's the best one could do - make a PR to make Ace do the ordering itself as long as the result set is < limit, but detecting a limit in the query itself could be tricky.

**Reply by joomlabeat** [reply]
Hi Jason! It does make sense.

I would also like to correct myself on what I wrote above. I just tested Sequel pro this morning for the feature I described. I really had the impression that it was possible to re-order a limited found set without making a second query from scratch (ordering only the existing returned results), but it seems that's not the case with Sequel pro either. 
I don't know how I got that impression.

I still feel that being able to sort an existing found-set could be handy and useful. For me, anything that allows me to easily visualise, validate and manipulate data is always a good thing.

I could imagine 2 approaches here:
 - Outside the SQL, by having the app sorting the found-set without making a new query
 - With SQL, by wrapping the written query inside a new query and applying the new order, something like:
 select * from (
	select id, title, date 
        from table
	order by id desc limit 10
	) as main
order by date DESC

Although, I don't know how something like this would scale with way more complex queries.

If such a feature should be added in the app, then maybe that should be adjustable through a setting - maybe a checkbox in the query editor tab that would enable/disable it.

Anyway just a few thoughts
